src/Integrations/Integrations/Laravel/LaravelIntegration.php

```diff
@@ -288,6 +311,125 @@ function ($exceptionHandler, $scope, $args) use ($rootSpan, $integration) {
             }
         );
 
+        // Used by Laravel >= 5.0
+        \DDTrace\hook_method(
+            'Illuminate\Auth\SessionGuard',
+            'attempt',
+            null,
+            function ($This, $scope, $args, $loginSuccess) use ($rootSpan, $integration) {
+                if ($loginSuccess) {
+                    return;
+                }
+
+                if (!function_exists('\datadog\appsec\track_user_login_failure_event'))
+                {
+                    return;
+                }
+                \datadog\appsec\track_user_login_failure_event(null, false, [], true);
+            }
+        );
+
+        // Used by Laravel >= 5.0
+        \DDTrace\hook_method(
+            'Illuminate\Auth\SessionGuard',
+            'setUser',
+            function ($This, $scope, $args) use ($rootSpan, $integration) {
+                if (!function_exists('\datadog\appsec\track_user_login_success_event'))
+                {
+                    return;
+                }
+                if (!isset($args[0])) {
+                    return;
+                }
+                $user = $args[0];
+                $authClass = 'Illuminate\Contracts\Auth\Authenticatable';
+                if (!$user || !($user instanceof $authClass)) {
+                    return;
+                }
+
+                $metadata = [];
+                if (isset($user['name'])) {
+                    $metadata['name'] = $user['name'];
+                }
+                if (isset($user['email'])) {
+                    $metadata['email'] = $user['email'];
+                }
+
+                \datadog\appsec\track_user_login_success_event($user->getAuthIdentifier(), $metadata, true);
+            }
+        );
+
+        // Used by Laravel < 5.0
+        \DDTrace\hook_method(
+            'Illuminate\Auth\Guard',
+            'setUser',
+            function ($This, $scope, $args) use ($rootSpan, $integration) {
+                if (!function_exists('\datadog\appsec\track_user_login_success_event'))
+                {
+                    return;
+                }
+                if (!isset($args[0])) {
+                    return;
+                }
+                $user = $args[0];
+                $authClass = 'Illuminate\Auth\UserInterface';
```

Comment:
I was wondering if this should not be rather this (with the `\` in the beginning):
```suggestion
                $authClass = '\Illuminate\Auth\UserInterface';
```
But there seems some magic going on ...
https://3v4l.org/G1UNq

In the function bellow (following the same schema), you where adding the `\` at the beginning, maybe we should do it here as well. Although it looks like its working, I would think it should not 😆 
The same applies to the function above this one

Reply:
Thanks for this. I will change it everywhere

Reply:
The better style would be rather the opposite - stringified names are always fully qualified. I fact `\Illuminate\Auth\UserInterface::class` will give you `'Illuminate\Auth\UserInterface'`.

Reply:
TIL

Reply:
ok, I will remove the initial `\` from all these places